crates/red_knot_python_semantic/resources/mdtest/scopes/global.md

```diff
@@ -0,0 +1,200 @@
+# `global` references
+
+## Implicit global in function
+
+A name reference to a never-defined symbol in a function is implicitly a global lookup.
+
+```py
+x = 1
+
+def f():
+    reveal_type(x)  # revealed: Unknown | Literal[1]
+```
+
+## Explicit global in function
+
+```py
+x = 1
+
+def f():
+    global x
+    reveal_type(x)  # revealed: Unknown | Literal[1]
+```
+
+## Unassignable type in function
+
+```py
+x: int = 1
+
+def f():
+    y: int = 1
+    # error: [invalid-assignment] "Object of type `Literal[""]` is not assignable to `int`"
+    y = ""
+
+    global x
+    # @Todo(error: [invalid-assignment] "Object of type `Literal[""]` is not assignable to `int`")
+    x = ""
+```
+
+## Nested intervening scope
+
+TODO this should give the outer module type of `Unknown | Literal[1]`, not `Literal[""]`
+
+```py
+x = 1
+
+def outer():
+    x = ""
```

Comment:
Nit: you could use declarations here (`x: int = 1` and `x: str = ""`) to get rid of the `Unknown |` in the types, which is not really relevant to the point of this test and kinda just noise? But it's fine either way.